crates/sui-core/src/validator_tx_finalizer.rs

```diff
@@ -170,6 +196,40 @@ where
         self.metrics.finalization_succeeded(start_time);
         Ok(true)
     }
+
+    // We want to avoid all validators waking up at the same time to finalize the same transaction.
+    // That can lead to a waste of resource and flood the network unnecessarily.
+    // Here we use the transaction digest to determine an order of all validators.
+    // The first few will wake up one by one with incremental delays to finalize the transaction.
+    // The rest will wake up at the same time afterward. The hope is that the first few should be
+    // able to finalize the transaction, and the rest will see it already executed and do not need
+    // to do anything.
+    fn determine_finalization_delay(
+        &self,
+        tx_digest: &TransactionDigest,
+    ) -> anyhow::Result<(usize, Duration)> {
+        let order = self
+            .agg
+            .committee
+            .shuffle_by_stake_from_tx_digest(tx_digest);
```

Comment:
Given that the order is on per transaction basis, there will be transactions whose first 5 are not part of the initial 2f+1, which causes the rest to execute the txn at the same time.

Therefore, I wonder if it makes sense to simply do incremental wait for all validators, or at least the first f+1?

Reply:
Good question! I was thinking that 1min should be sufficient for finality + checkpoint execution in most of the cases?

Reply:
Another thing worth mentioning is that in theory a certificate should be broadcasted to all validators, even though a FN only needs 2f+1 responses. The only issue here would be that if the 5 validators are all slow validators that are behind be more than 1 minute.